persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -187,271 +190,268 @@ protected RocksDB openDatabase(Path location, Options options) throws RocksDBExc
-    public void destroy() {
-        stop();
-        Util.recursiveFileRemove(getLocation().toFile());
-        Util.recursiveFileRemove(getExpirationLocation().toFile());
+        }, "rocksdb-stop");
     }
 
     @Override
-    public boolean isAvailable() {
-        return getLocation().toFile().exists() && getExpirationLocation().toFile().exists();
+    public Set<Characteristic> characteristics() {
+        return EnumSet.of(Characteristic.BULK_READ, Characteristic.EXPIRATION, Characteristic.SEGMENTABLE);
     }
 
     @Override
-    public void clear() {
-        handler.clear(null);
+    public CompletionStage<Boolean> isAvailable() {
+        return blockingManager.supplyBlocking(() -> getLocation().toFile().exists() && getExpirationLocation().toFile().exists(),
+                "rocksdb-available");
     }
 
     @Override
-    public void clear(IntSet segments) {
-        handler.clear(segments);
+    public CompletionStage<Void> clear() {
+        return handler.clear();
     }
 
     @Override
-    public int size() {
-        return handler.size(null);
-    }
-
-    @Override
-    public int size(IntSet segments) {
+    public CompletionStage<Long> size(IntSet segments) {
         return handler.size(segments);
     }
 
     @Override
-    public boolean contains(Object key) {
-        return handler.contains(-1, key);
+    public CompletionStage<Long> approximateSize(IntSet segments) {
+        return handler.approximateSize(segments);
     }
 
     @Override
-    public boolean contains(int segment, Object key) {
-        return handler.contains(segment, key);
-    }
-
-    @Override
-    public Publisher<K> publishKeys(Predicate<? super K> filter) {
-        return handler.publishKeys(null, filter);
+    public CompletionStage<Boolean> containsKey(int segment, Object key) {
+        // This might be able to use RocksDB#keyMayExist - but API is a bit flaky
+        return load(segment, key)
+                .thenApply(Objects::nonNull);
     }
 
     @Override
     public Publisher<K> publishKeys(IntSet segments, Predicate<? super K> filter) {
-        return handler.publishKeys(segments, filter);
-    }
-
-    @Override
-    public Publisher<MarshallableEntry<K, V>> entryPublisher(Predicate<? super K> filter, boolean fetchValue, boolean fetchMetadata) {
-        return handler.publishEntries(null, filter, fetchValue, fetchMetadata);
+        return Flowable.fromPublisher(handler.publishEntries(segments, filter, false))
+                .map(MarshallableEntry::getKey);
     }
 
     @Override
-    public Publisher<MarshallableEntry<K, V>> entryPublisher(IntSet segments, Predicate<? super K> filter,
-                                                             boolean fetchValue, boolean fetchMetadata) {
-        return handler.publishEntries(segments, filter, fetchValue, fetchMetadata);
+    public Publisher<MarshallableEntry<K, V>> publishEntries(IntSet segments, Predicate<? super K> filter, boolean includeValues) {
```

Comment:
I missed this before, but I don't think `boolean includeValues` should be part of the `publishEntries` parameters. If you don't require the value, then you should just utilise `publishKeys`.

Probably irrelavant now, but `boolean includeValues` has no Javadocs in the interface.

Reply:
You can use publishEntries with includeValues as false to get the metadata still (although  most stores may still return the value).

Reply:
Ok, in that case `publishEntries` is still missing the JavDoc for the `includeValues` parameter. As part of the main javadocs "body" I think you should include your explanation of why `!includeValues` is useful and that some stores may still return the value in `MarshallableEntry`.